posthog/renderers.py

```diff
@@ -0,0 +1,46 @@
+import math
+from typing import Dict, Tuple
+from rest_framework.renderers import JSONRenderer
+
+CleaningMarker = bool | Dict[int, "CleaningMarker"]
+
+
+def clean_data_for_json(data) -> Tuple[CleaningMarker, CleaningMarker]:
+    """Replace NaNs and Infinities with None, in-place, marking which fields had to be scrubbed.
+    Return markers that should be set at the parent level, which is used for setting the markers recursively."""
+    if isinstance(data, float):
+        return (math.isnan(data), math.isinf(data))
```

Comment:
That is true – this is pretty redundant for scalars, but this function also marks nested nans/infs, which do need separate collection. Because this is recursive, the property of always returning a tuple makes this more ergonomic

Reply:
Makes sense 👍